compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -40,71 +36,43 @@ namespace mlir::iree_compiler {
 #define GEN_PASS_DEF_GPUMATERIALIZEDEVICEENCODINGPASS
 #include "iree/compiler/Codegen/Common/GPU/Passes.h.inc"
 
-/// Returns the corresponding native vector sizes defined by the `mma`
-/// intrinsic.
-static SmallVector<int64_t> getIntrinsicVectorSize(IREE::GPU::MMAIntrinsic mma,
-                                                   int64_t roleIdx) {
+static std::optional<MaterializeEncodingInfo::Swizzle>
+getSwizzle(IREE::GPU::MMAIntrinsic mma, int operandIdx) {
```

Comment:
I think this should depend on `IREE::GPU::DataTiledMMAAttr` instead of `IREE::GPU::MMAIntrinsic`, right?

Reply:
Oh, I forgot to add a comment explaining that.
The idea is that this is for now not actually supporting unrolling. This is just a database of the layout of each intrinsic. In a follow-up when we start supporting unrolling, we will add a helper function to infer the unrolled kernel layout from that. This will allow us to freely play with various unrolling values while letting that logic infer the kernel layouts.

Reply:
Ok, that sounds good to me. As long as the `MaterializeEncodingInfo` returned by `materializeEncodingForTarget` is going to have the full swizzle including unrolling, then this looks good.